fix(root): change proptable type to fix build

GCHQ-Developer-299 opened the pull request:
build was failing because of type mismatch, have typed propData as stencilProp

GCHQ-Developer-741 commented:
Just a question, but the error seemed to suggest that there was a problem because `complexType` could be undefined, but we were treating it as though it wasn't. Does this change actually address this possibility? The way I read it is that the error is being suppressed but we could still get a runtime error if the docs is missing the `complexType` property.

GCHQ-Developer-741 commented:
> Just a question, but the error seemed to suggest that there was a problem because `complexType` could be undefined, but we were treating it as though it wasn't. Does this change actually address this possibility? The way I read it is that the error is being suppressed but we could still get a runtime error if the docs is missing the `complexType` property and then tries to access the resolved and original values

Another note, but not sure why we even have a StencilProp type at all, given the the slot, method, etc components all import the type from the JSONDocs package. Seems like unnecessary duplication which caused this error to sneak through in the first place
![Screenshot from 2024-05-30 16-15-48](https://github.com/mi6/ic-design-system/assets/117368893/edae0c84-6507-4a40-9d3d-30d44fd0746f)

GCHQ-Developer-299 commented:
> > Just a question, but the error seemed to suggest that there was a problem because `complexType` could be undefined, but we were treating it as though it wasn't. Does this change actually address this possibility? The way I read it is that the error is being suppressed but we could still get a runtime error if the docs is missing the `complexType` property and then tries to access the resolved and original values
> 
> Another note, but not sure why we even have a StencilProp type at all, given the the slot, method, etc components all import the type from the JSONDocs package. Seems like unnecessary duplication which caused this error to sneak through in the first place ![Screenshot from 2024-05-30 16-15-48](https://private-user-images.githubusercontent.com/117368893/335262797-edae0c84-6507-4a40-9d3d-30d44fd0746f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNDYyMzYsIm5iZiI6MTcxNzE0NTkzNiwicGF0aCI6Ii8xMTczNjg4OTMvMzM1MjYyNzk3LWVkYWUwYzg0LTY1MDctNGE0MC05ZDNkLTMwZDQ0ZmQwNzQ2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUzMVQwODU4NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNGM3MzZiNWExNmZjY2MwMTljNDE0ZDg0OWViNGViYWM5NmJjMTBlZDVlOTc3ODg2YzEzOTViMjFjZTRhNTkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6ozQTAybcuTySGVuUi7YhOJq_jgY_r_t9xdeTSos0A0)

Good spot! I've removed StencilProp and added logic to check for types :)
